File: src/main/java/org/soraworld/lime/Limitime.java
package org.soraworld.lime;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.plugin.java.JavaPlugin;
import org.soraworld.lime.command.CommandLimit;
import org.soraworld.lime.command.IICommand;
import org.soraworld.lime.config.Config;
import org.soraworld.lime.listener.EventListener;
import org.soraworld.lime.util.ListUtils;

public class Limitime extends JavaPlugin {

    private Config config;
    private IICommand command;

    @Override
    public void onEnable() {
        config = new Config(this.getDataFolder());
        config.load();
        config.save();
        this.getServer().getPluginManager().registerEvents(new EventListener(config), this);
        command = new CommandLimit("limit", this, config);
    }

    @Override
    public void onDisable() {
        config.save();
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        return command.execute(sender, ListUtils.arrayList(args));
    }

}

File: src/main/java/org/soraworld/lime/Limitime.kt
package org.soraworld.lime

import org.soraworld.lime.command.CommandLimit
import org.soraworld.lime.config.LimitConfig
import org.soraworld.lime.constant.Constant
import org.soraworld.lime.listener.EventListener
import org.soraworld.violet.VioletPlugin
import org.soraworld.violet.command.IICommand
import org.soraworld.violet.config.IIConfig
import java.io.File

class Limitime : VioletPlugin() {

    override fun registerCommand(): IICommand? {
        return if (iconfig is LimitConfig) CommandLimit(Constant.PLUGIN_ID, Constant.PERM_ADMIN, iconfig as LimitConfig)
        else null
    }

    override fun registerConfig(path: File): IIConfig {
        return LimitConfig(path)
    }

    override fun registerEvents() {
        if (iconfig is LimitConfig) registerEvent(EventListener(iconfig as LimitConfig))
    }

    override fun afterEnable() {}

    override fun beforeDisable() {}

}
